### PR TITLE
Allow overwriting media files of dependencies

### PR DESCRIPTION
**Pull request by Desour**
- Achieves the same as #10449, but doesn't do weird string compares.
- `getModsMediaPaths` was changed to first add the paths of the depender mods before the dependees mods' paths.
- The only other place where `getModsMediaPaths` is used is in a unit test, and all unit tests pass.
- devtest now uses the now-official (but discouraged) mechanic of overwriting media.
- doc is updated acoringly.

## To do

This PR is a Ready for Review.

## How to test

Start devtest and look at dirt.

### CODE REVIEW

File: src/server/mods.cpp [hunk 1]
```diff
@@ -98,11 +99,17 @@ void ServerModManager::getModNames(std::vector<std::string> &modlist) const
...
-		fs::GetRecursiveDirs(paths, spec.path + DIR_DELIM + "sounds");
-		fs::GetRecursiveDirs(paths, spec.path + DIR_DELIM + "media");
-		fs::GetRecursiveDirs(paths, spec.path + DIR_DELIM + "models");
-		fs::GetRecursiveDirs(paths, spec.path + DIR_DELIM + "locale");
-	}
+	std::for_each(m_sorted_mods.rbegin(), m_sorted_mods.rend(),
+			[&paths](const ModSpec &spec) {
+				fs::GetRecursiveDirs(paths,
+						spec.path + DIR_DELIM + "textures");
+				fs::GetRecursiveDirs(
```
Comment (sfan5) [inline src/server/mods.cpp:106]: this is a bit ... ugly

I'd prefer manual iterator usage over this.

Comment (Desour) [reply src/server/mods.cpp]: Done.

**Review by sfan5 — APPROVED** [review]
LGTM

File: src/server/mods.cpp [hunk 2]
```diff
@@ -24,6 +24,7 @@ with this program; if not, write to the Free Software Foundation, Inc.,
 #include "content/subgames.h"
 #include "porting.h"
 #include "util/metricsbackend.h"
+#include <algorithm>
```
Comment (sfan5) [inline src/server/mods.cpp:27]:
```suggestion
```

**Review by appgurueu — APPROVED** [review]
Looks good. LGTM.

**Comment by appgurueu** [comment]
This should close #10936 ?

File: src/server/mods.cpp [hunk 3]
```diff
@@ -98,7 +98,8 @@ void ServerModManager::getModNames(std::vector<std::string> &modlist) const
 
 void ServerModManager::getModsMediaPaths(std::vector<std::string> &paths) const
 {
-	for (const ModSpec &spec : m_sorted_mods) {
+	for (auto it = m_sorted_mods.crbegin(); it != m_sorted_mods.crend(); it++) {
```
Comment (rubenwardy) [inline src/server/mods.cpp:101]:
```suggestion
	// See doxygen comment in header, iterate in reverse to apply texture
	// overrides properly.		
	for (auto it = m_sorted_mods.crbegin(); it != m_sorted_mods.crend(); it++) {
```

**Review by nerzhul — APPROVED** [review]
lgtm

**Review by rubenwardy — APPROVED** [review]

**Pull request merged**